sync/phabricator/listen.py

```diff
@@ -0,0 +1,121 @@
+import time
+import re
+from phabricator import Phabricator
+
+from .. import log
+from ..tasks import handle
+
+
+logger = log.get_logger(__name__)
+
+
+class PhabEventListener:
+
+    blacklist = ["added inline comments to D",
+                 "added a comment to D",
+                 "added a reviewer for D",
+                 "removed a reviewer for D",
+                 "requested review of D",
+                 "requested changes to D",
+                 "added a subscriber to D",
+                 "added a project to D",
+                 "committed rMOZILLA",
+                 "committed rVCT",
+                 "committed rCIADMIN",
+                 "edited reviewers for D",
+                 "updated the summary of D",  # Maybe useful to upstream info?
+                 "accepted D",  # Maybe useful to upstream info?
+                 "retitled D",  # Maybe useful to upstream info?
+                 "blocking reviewer(s) for D",
+                 "planned changes to D",
+                 "resigned from D"]
+
+    event_mapping = {
+        "updated the diff for D": "commit",
+        "created D": "commit",
+        "closed D": "closed",
+        "abandoned D": "closed",  # Bit unsure about this one
+    }
+
+    def __init__(self, config):
+        self.running = True
+        self.timer_in_seconds = config['phabricator']['listener']['interval']
+        self.latest = None
+
+        self.phab = Phabricator(host='https://phabricator.services.mozilla.com/api/',
+                                token=config['phabricator']['token'])
+        self.phab.update_interfaces()
+
+    def run(self):
+        # Run until told to stop.
+        while self.running:
+            feed = self.get_feed()
+            self.parse(feed)
+            time.sleep(self.timer_in_seconds)
+
+    def get_feed(self, before=None):
+        """ """
+        if self.latest and before is None:
+            before = int(self.latest['chronologicalKey'])
+
+        done = False
+        feed = []
+
+        def chrono_key(feed_story_tuple):
+            return int(feed_story_tuple[1]["chronologicalKey"])
+
+        # keep fetching stories from Phabricator until there are no more stories to fetch
+        while not done:
```

Comment:
I think this might be clearer with a `while True:/break` setup, but I'm not sure.

Reply:
I'll look into alternative setups